src/Api/EncodedStringTest.cpp

```diff
@@ -0,0 +1,521 @@
+                       0x11};
+
+  EXPECT_EQ(expected, decoded);
+}
+
+TEST(EncodedString, Decode80ByteStringIncluding16AdditionalCharacters) {
+  std::vector<uint64_t> additional_char;
+  additional_char.push_back(0x8877665544332211);
+  additional_char.push_back(0x8877665544332211);
+  std::string decoded =
+      DecodeString(0x8877665544332211, 0x8877665544332211, 0x8877665544332211, 0x8877665544332211,
+                   0x8877665544332211, 0x8877665544332211, 0x8877665544332211, 0x8877665544332211,
+                   additional_char.data(), additional_char.size());
+  std::string expected{
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 1
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 2
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 3
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 4
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 5
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 6
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 7
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 8
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 9
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 10
+  };
+
+  EXPECT_EQ(expected, decoded);
+}
+
+TEST(EncodedString, EncodeEmptyString) {
+  ApiScopeStart encoded_string;
+  EncodeString("", &encoded_string);
+  EXPECT_EQ(encoded_string.encoded_name_1(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_2(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_3(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_4(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_5(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_6(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_7(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_8(), 0);
+  EXPECT_THAT(encoded_string.encoded_name_additional(), ElementsAre());
+}
+
+TEST(EncodedString, Encode1ByteString) {
+  ApiScopeStart encoded_string;
+  std::string decoded_string;
+  decoded_string.push_back(0x11);
+  EncodeString(decoded_string.c_str(), &encoded_string);
+  EXPECT_EQ(encoded_string.encoded_name_1(), 0x0000000000000011);
+  EXPECT_EQ(encoded_string.encoded_name_2(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_3(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_4(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_5(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_6(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_7(), 0);
+  EXPECT_EQ(encoded_string.encoded_name_8(), 0);
+  EXPECT_THAT(encoded_string.encoded_name_additional(), ElementsAre());
+}
+
+TEST(EncodedString, Encode4ByteString) {
+  ApiScopeStart encoded_string;
+  std::string decoded_string;
+  decoded_string.push_back(0x11);
+  decoded_string.push_back(0x22);
+  decoded_string.push_back(0x33);
+  decoded_string.push_back(0x44);
```

Comment:
Here and below, why not use the constructor like done in the tests above?

Reply:
Fair enough. Done.